misc/docker-ci/build

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+# To build Docker images for CI and push the artifacts to Docker Hub.
+# https://hub.docker.com/r/h2oserver/h2o-ci
+#
+# usage: misc/docker-ci/build variants...
+#        variants might be ubuntu1604 ubuntu2004 (corresponding misc/docker-ci/Dockerfile.$variant)
+
+set -xe
+
+IMAGE_NAME=h2oserver/h2o-ci
+
+SCRIPT_DIR=$(realpath $(dirname $0))
+PROJECT_ROOT_DIR=$(realpath "${SCRIPT_DIR}/../..")
+
+# The build directory must be empty to avoid include files which are should not be published.
+BUILD_DIR="$SCRIPT_DIR/.empty"
+mkdir -p "$BUILD_DIR"
+trap 'rm -rf "${BUILD_DIR}"' 0
+cd $BUILD_DIR
```

Comment:
I think we need to bail out if this fails. Otherwise, we might build and push a docker image that contains sensitive information.

Reply:
Yep. `set -e` is set to bail out on failing.

Reply:
Ah, I missed that! Thanks.